File: 7_kyu/Right_in_the_Centre.py
def is_in_middle(sequence: str) -> bool:
    SEQ="abc"
    if len(sequence) < len(SEQ):
        return False
    start,d = divmod(len(sequence)-len(SEQ), 2)
    return sequence[start:start+len(SEQ)] == SEQ or sequence[start+d:start+d+len(SEQ)] == SEQ

File: 7_kyu/Right_in_the_Centre.rs
// Returns true if "abc" is in the middle of seq.
// Middle is interpreted on bytes, not on chars.
fn is_in_middle(seq: &str) -> bool {
    const SEARCH: &str = "abc";
    if seq.len() < SEARCH.len() {
        return false;
    }
    let option1: usize = (seq.len() + 1 - SEARCH.len()) / 2;
    let option2: usize = (seq.len() - SEARCH.len()) / 2;
    &seq[option1..option1 + 3] == SEARCH || &seq[option2..option2 + 3] == SEARCH
}
